### ISSUE TITLE
"implementation is left-biased" - but fromList is not

### ISSUE THREAD

**Issue by jwaldmann** [haskell/containers#333]
The actual behaviour of `fromList` surprised me, given the general "left-biased" feeling of the library.
I think I was expecting that these be equivalent
```
fromListWith (++) [ (0,"foo"), (1,"bar"), (0,"oof") ]
unionsWith (++) [ singleton 0 "foo", singleton 1 "bar", singleton 0 "oof" ]
```
but they are not. Perhaps add a warning in the documentation of `fromListWith`? Currently, you have to parse the example to infer the semantics.

**Comment by treeowl** [comment]
Yes, this trips me up too. The idea is that it does a left fold with
insert, which indeed is unexpected. Feel free to open a PR, or I'll try to
fix it myself.

On Sep 4, 2016 5:32 AM, "jwaldmann" <notifications@github.com> wrote:

The actual behaviour of fromList surprised me, given the general
"left-biased" feeling of the library.
I think I was expecting that these be equivalent

fromListWith (++) [ (0,"foo"), (1,"bar"), (0,"oof") ]
unionsWith (++) [ singleton 0 "foo", singleton 1 "bar", singleton 0 "oof" ]

but they are not. Perhaps add a warning in the documentation of fromListWith?
Currently, you have to parse the example to infer the semantics.

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub
<https://github.com/haskell/containers/issues/333>, or mute the thread

**Comment by sjakobi** [comment]
Related: https://github.com/haskell-unordered-containers/unordered-containers/issues/264
